app/core/utils.py

```diff
@@ -197,3 +202,144 @@ def get_gpu_info():
     mem_percentage = (used_memory_value / total_memory_value) * 100
 
     return gpu_name, total_memory_value, used_memory_value, mem_percentage
+
+
+def extract_labels_from_html_file(html_content, class_names):
+    soup = BeautifulSoup(html_content, "html.parser")
+    labels = soup.select(class_names)
+    return [label.get_text() for label in labels]
```

Comment:
much generator such efficiency

```suggestion
    return (label.get_text() for label in labels)
```